### PR TITLE
[WIP] Fixed test failing with npm command

### PR DESCRIPTION
**Pull request by viliusl**
Ok, so there are several issues with this particular e2e test:
 - if you install via npm or yarn you get `web-driver-manager` installed in different locations under `node_modules` - yarn is more aggressive in flattening deps whereas npm is more lax. For this I just search `node_modules` for `chromedriver_2.28.zip`;
 - when you have auth in your `.npmrc` (ex. `//registry.npmjs.org/:_authToken=...`) and artifactory as your repo, yarn has a [bug](https://github.com/yarnpkg/yarn/issues/2554) where it sends auth for any scoped packages (ex. `@types`) and artifactory fails with 401 (as auth token is not against artifactory but npmjs registry). when you installed yoshi with yarn, packages are already cached and you don't see this error.

I cannot verify this against tc for obvious reasons, but locally it passes and so should travis.

### CODE REVIEW

File: yoshi/test/helpers/test-phases.js [hunk 1]
```diff
@@ -18,8 +18,10 @@ class Test {
   }
 
   setup(tree, hooks = []) {
+    //fix for ci with private registries (artifactory) where existing auth is used for all scoped dependencies
+    const treeWithDefaults = Object.assign({'.yarnrc': '"@types:registry" "https://registry.npmjs.org/"'}, tree);
```
Comment (tqmukas) [inline yoshi/test/helpers/test-phases.js]: I think this is bad idea. `test-phases.js` should not be coupled to neither `npm` neither `yarn`. Please think about other solution to solve this.

Comment (viliusl) [inline yoshi/test/helpers/test-phases.js]: @tqmukas - this is due to bug in `yarn` and our registry set-up so of course this is a work-around. We could export this to ci agent configs, but then on next `@x` usage we would have to go there and update separately. This one at least gives us visibility.

Cannot think of better way to solve it:/

**Comment by ronami** [comment]
@tqmukas what's the status with the PR?

**Comment by viliusl** [comment]
@tqmukas failing - will investigate it, but fails only on travis for now:)

**Comment by viliusl** [comment]
closing for now esp as we migrated all builds to yarn

**Pull request closed without merging**